salt/modules/state.py

```diff
@@ -2286,7 +2288,85 @@ def _disabled(funs):
     return ret
 
 
-def event(tagmatch='*',
+@with_deprecated(globals(), "Fluorine")
+def event(tagmatch='.*',
+          count=-1,
+          quiet=False,
+          sock_dir=None,
+          pretty=False,
+          node='minion'):
+    r'''
+    Watch Salt's event bus and block until the given tag is matched
+
+    .. versionadded:: 2016.3.0
+
+    This is useful for utilizing Salt's event bus from shell scripts or for
+    taking simple actions directly from the CLI.
+
+    Enable debug logging to see ignored events.
+
+    :param tagmatch: the event is written to stdout for each tag that matches
+        this regular expression.
+    :param count: this number is decremented for each event that matches the
+        ``tagmatch`` parameter; pass ``-1`` to listen forever.
+    :param quiet: do not print to stdout; just block
+    :param sock_dir: path to the Salt master's event socket file.
+    :param pretty: Output the JSON all on a single line if ``False`` (useful
+        for shell tools); pretty-print the JSON output if ``True``.
+    :param node: Watch the minion-side or master-side event bus.
+
+    CLI Example:
+
+    .. code-block:: bash
+
+        salt-call --local state.event pretty=True
+    '''
+    try:
+        patt = re.compile(tagmatch)
+    except:
+        raise SaltInvocationError(
+            'Invalid regular expression "{0}" given as parameter "tagmatch"'.format(tagmatch)
+        )
+
+    sevent = salt.utils.event.get_event(
+            node,
+            sock_dir or __opts__['sock_dir'],
+            __opts__['transport'],
+            opts=__opts__,
+            listen=True)
+
+    we_are_counting = True if count > -1 else False
+
+    while True:
```

Comment:
What happens if `count == -1`?

Reply:
If you would write a unit test for this thing, you would quickly find that it contains this bug. :wink: You can look at [`test_ansiblegate.py`](https://github.com/samilaine/salt/blob/develop/tests/unit/modules/test_ansiblegate.py) as a reference how to write them.

Generally, you can just run them with `pytest`, like so:

```
$ cd $GIT/salt/tests/unit/modules
$ pytest test_state.py
```

Just add your test(s) to the `test_state.py` file.